src/app/src/api/submissions.js

```diff
@@ -0,0 +1,52 @@
+import api from './api';
+import TAGS, {
+    getListTagProvider,
+    getNewItemTagInvalidator,
+    getSingleItemProvider,
+    getUpdateItemTagInvalidator,
+} from './tags';
+
+const submissionApi = api.injectEndpoints({
+    endpoints: build => ({
+        getDraftSubmissions: build.query({
+            query: () => ({
+                url: '/submissions',
+                method: 'GET',
+                params: { type: 'drafts' },
+            }),
+            providesTags: getListTagProvider(TAGS.SUBMISSION),
+        }),
+
+        getSubmissionDetails: build.query({
+            query: id => ({
+                url: `/submissions/${id}`,
+                method: 'GET',
+            }),
+            providesTags: getSingleItemProvider(TAGS.SUBMISSION),
+        }),
+
+        startNewSubmission: build.mutation({
+            query: newSubmission => ({
+                url: '/submissions',
+                method: 'POST',
+                body: newSubmission,
+            }),
+            invalidatesTags: getNewItemTagInvalidator(TAGS.SUBMISSION),
+        }),
+
+        submitSubmission: build.mutation({
+            query: id => ({
+                url: `/submissions/submit/${id}`,
+                method: 'POST',
+            }),
+            invalidatesTags: getUpdateItemTagInvalidator(TAGS.SUBMISSION),
+        }),
+    }),
+});
+
+export const {
+    useGetDraftSubmissionsQuery,
```

Comment:
Would be curious for a brief demo of how this hook gets used

Reply:
[`useMutation`](https://redux-toolkit.js.org/rtk-query/api/created-api/hooks#usemutation) is slightly different in it's usage:

```jsx
function NewDraftPage() {
    const [formValues, setFormValues] = useState({});
    const [createNewDraft, { isLoading }] = useCreateNewDraftMutation();

    function handleCreateNewDraft() {
        createNewDraft(formValues);
    }

   if (isLoading) {
        return <CircularProgress />
    } 

    return (<Form>
        <Button onClick={handleCreateNewDraft}>Create New Draft</Button>
    </Form>);
}